### PR TITLE
Allow SeamlessImmutable/ImmutableArray.concat to accept an array argument

### PR DESCRIPTION
**Pull request by rharriso**
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes:
```javascript
const a0 = SeamlessImmutable([1,2,3,4])
const a1 = SeamlessImmutable([6,7,8,9])
a0.concat(5).concat(a1)
// [ 1, 2, 3, 4, 5, 6, 7, 8, 9 ]
```
- [x] Increase the version number in the header if appropriate.
- [N/A ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

File: types/seamless-immutable/index.d.ts [hunk 1]
```diff
@@ -1,4 +1,4 @@
-// Type definitions for Seamless-immutable 7.1
+// Type definitions for Seamless-immutable 7.2
```
Comment (RMHonor) [inline types/seamless-immutable/index.d.ts]: Seamless Immutable is still on version 7.1.x, best to leave this as it was (@latest = 7.1.4)

**Review by RMHonor — CHANGES_REQUESTED** [review]

**Comment by rharriso** [comment]
Updated and the Ci is passing

**Review by RMHonor — APPROVED** [review]

**Comment by uniqueiniquity** [comment]
Closing to rerun CI.

**Comment by uniqueiniquity** [comment]
Congratulations on your first DefinitelyTyped contribution!
Thank you for being a part of the community!

**Pull request merged**